sequence_task_1/student/include/impl.h

```diff
@@ -3,38 +3,85 @@
 #include <string>
 #include <stdexcept>
 
+using size_type = size_t;
+
 template<class Container, class Song_t>
 class StaticPlaylist
 {
 public:
     /** @todo Member traits */
-
+    using value_type = typename Container::value_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
     /** @todo Iterators */
+    const_iterator begin() const {
+        return m_tracklist.cbegin();
+    }
 
-    StaticPlaylist() = default;
+    const_iterator end() const {
+        return m_tracklist.cend();
+    }
 
+    StaticPlaylist() : m_tracklist() {}
     /** @todo Constructor from any reversible sequence container */
-
+    template<class PlayList>
+    StaticPlaylist(PlayList& playList)
```

Comment:
1. You don't modify the playlist, so you should not take the non-const reference to it
2. You may use the constructor with two iterators in the initialization list